### DISCUSSION TITLE
Set gaps for specific workspace

### DISCUSSION THREAD

**Discussion by RiccardoPP** [hyprwm/Hyprland#1345] [Q&A] [answered]
Hello and thanks for Hyprland! 

I'd like to have a keybind to toggle no_gaps in a workspace (with at least two windows inside). Is it possible? I've searched in the wiki with no success.

Thank you.

**Comment by vaxerski** [comment]
no clue what you mean by that but it should be scriptable

**Reply by RiccardoPP** [reply]
Thanks for the reply.

Usually I have a certain gaps_in and gaps_out set for aesthetic but I'd like to have the ability to set both to 0 only in the active workspace to utilize as much screen as possible (for ex: when I'm working with VSCode and Firefox side by side).

**Reply by vaxerski** [reply]
yeah you can script that pretty easily

**Reply by levnikmyskin** [reply]
Hi! Would you mind sharing how to do this (if you managed)? @RiccardoPP

**Reply by RiccardoPP** [reply]
Hello, check the bind rule in my other message.

`bind=SUPER,G,exec,hyprctl --batch "keyword general:gaps_in 0 ; keyword general:gaps_out 0 ; keyword general:border_size 0 ; keyword decoration:rounding 0 ; keyword decoration:drop_shadow false"`

**Reply by levnikmyskin** [reply]
Thanks for your reply! And yes, I saw the hyprctl command. I was wondering how to set it for a specific window/workspace only, cannot seem to find how on the wiki

**Reply by RiccardoPP** [reply]
Yes, could be great, but I've not found how to do it.

**Reply by levnikmyskin** [reply]
Mh yeah same. I guess one way could be to write a little script which listens for [events](https://wiki.hyprland.org/IPC/), and toggle the no gaps/no borders etc. when, e.g., you have a specific window on a specific workspace or something like that.

I believe it would be cool to have workspace-specific configurations though (or even, window-specific gaps and borders)

**Comment by RiccardoPP** [comment]
Ok, to set gaps to 0, remove rounding, remove border, disable shadow:

```
bind=SUPER,G,exec,hyprctl --batch "keyword general:gaps_in 0 ; keyword general:gaps_out 0 ; keyword general:border_size 0 ; keyword decoration:rounding 0 ; keyword decoration:drop_shadow false"
```

Now I need to activate the hyprctl command only in the active workspace, or to create a static workspace with this specific gaps.

**Comment by levnikmyskin** [comment, marked as answer]
This is now merged. See #1986 and the [Hyprland wiki](https://wiki.hyprland.org/Configuring/Workspace-Rules/).

**Reply by RiccardoPP** [reply]
> This is now merged. See #1986 and the [Hyprland wiki](https://wiki.hyprland.org/Configuring/Workspace-Rules/).

Great, thanks!
